### ISSUE TITLE
Enable VP9 support on Firefox

### ISSUE THREAD

**Issue by bt90** [jitsi/jitsi-meet#10657]
**Is your feature request related to a problem you are facing?**

Currently VP9 support is opt-in and only works with Chrome. If one participant joins with Firefox, all clients switch to VP8 (see https://github.com/jitsi/jitsi-meet/pull/9552). We'd like to use VP9 but this is currently not possible in all scenarios as we have a mixed Chrome/Firefox environment.

**Describe the solution you'd like**

It would be nice to have VP9 available on Firefox in order to have a better video quality while keeping bandwidth low.

I assume that this is currently blocked by some upstream Firefox bug but i couldn't find any useful information apart from the PR above.

**Comment by bt90** [comment]
Ping

**Comment by bt90** [comment]
Blocked by https://bugzilla.mozilla.org/show_bug.cgi?id=1492500

**Comment by BinaryWizard904** [comment]
The Firefox exclusion of VP9 (described in #9552) is itself buggy and does not always act as @bt90 describes.  If you configure the `preferredCodec; 'VP9'` and `enforcePreferredCoded: true` in the config file, then there are circumstances where Jitsti will use VP9 in Firefox.

- When in a mixed Firefox/Chromium meeting, the first client to send video determines which codec is used for all subsequent videos.
  - If Chromium is the first to send video everyone uses VP9 going forwards.
  - If Firefox is the first everyone uses VP8 going forwards.

More interestingly, I haven't seen any issues with VP9 in Firefox 102.0.1 when I manage to get Jitsi to use it.  Video quality is both smooth and much improved over VP8 (regardless of whether it is being sent by Firefox or Chrome).  No freezing, crashing, corruption, etc.  Wonder if the Firefox issues were quietly fixed...

**Comment by bt90** [comment]
It appears to be working. Just tested with Firefox 102 using a modified useragent and the Jitsi app.

**Comment by saghul** [comment]
Ping @jallamsetty1 @nils-ohlmeier

**Comment by BinaryWizard904** [comment]
Looks like the code here is what's holding Firefox back.

https://github.com/jitsi/lib-jitsi-meet/blob/3cc1e01027a9a8b23d25959c096f3cc6e9f47ac2/modules/browser/BrowserCapabilities.js#L232-L234

Any word on whether VP9 is going to be enabled on Firefox or not?

**Comment by jallamsetty1** [comment]
The last time we tested, more than a month back, there were still freezes happening on the render side and on the encode side, Firefox would only send low resolutions because of low BWE (estimate starts higher but keeps dropping). I haven't tested with the latest Firefox release though.

**Comment by bt90** [comment]
Is there any news on this?

**Comment by bt90** [comment]
The upcoming Firefox 106 comes with libwebrtc 103. Maybe a good time to test this again :)

**Comment by paweldomas** [comment]
I'm not part of the core Jitsi team anymore, but if you want to see it sooner I'd suggest doing some testing yourself and posting the results here

**Comment by bt90** [comment]
Simulcast and SVC are still buggy.

https://bugzilla.mozilla.org/show_bug.cgi?id=1633876

**Comment by pkomisar** [comment]
Apparently p2p settings are what matter. I've been setting `videoQuality.preferredCodec` and `videoQuality.enforcePreferredCodec `but eventually what worked for me was setting
`p2p.preferredCodec='VP9'`

**Comment by saghul** [comment]
@jallamsetty1 can this be closed?

**Comment by bt90** [comment]
As the Firefox bug is still open, I don't think so.
